src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public void add(T value) {
+    private static final int ZERO_SIZE = 0;
+    private static final int HALF_SIZE = 2;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
```

Comment:
Make `Node<T>` an inner class. Inner classes should be placed at the end of the class they belong to.

Reply:
fix comment